.github/actions/action.yml

```diff
@@ -0,0 +1,24 @@
+name: uploadSizeArtifact
+description: 'Uploads a file using the commit hash as the name, and containing only the size of the Demo app in kB'
+inputs:
+  sha:
+    description: 'sha of the commit to sync, build, and upload an artifact for the size of'
+    required: true
+    type: string
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3 # Checkout repo at sha
+      with:
+        ref: ${{ inputs.sha }}
+    - name: Switch to current version of Xcode
+      run: scripts/xcode_select_current_version.sh
+    - name: scripts/xcodebuild_wrapper.sh 'ios_device_build Demo.Development Release build -archivePath $GITHUB_WORKSPACE/sha archive' # Build and archive release device build
```

Comment:
if this is in a different yml, can we just utilize the existing build ci and push a new artifact? rather than doing another build?

Reply:
only the head build will be needed

Reply:
I'm not sure there's a way to just add the action on a single build from the matrix?